src/tcgwars/logic/impl/gen4/RisingRivals.groovy

```diff
@@ -952,17 +1252,25 @@ public enum RisingRivals implements LogicCardInfo {
           move "Pick and Collect", {
             text "Search your discard pile for up to 4 basic Energy cards, show them to your opponent, and put them into your hand."
             energyCost ()
-            attackRequirement {}
+            attackRequirement {
+              assert my.discard.filterByType(BASIC_ENERGY) : "You have no basic Energy cards in your discard pile"
+            }
             onAttack {
-              damage 0
+              my.discard.select(max:4,"Search your discard pile for up to 4 basic Energy cards",cardTypeFilter(BASIC_ENERGY)).showToOpponent("Pick and Collect: Selected cards").moveTo(my.hand)
```

Comment:
Make sure it has a minimum of 1, once you declare using the ability, you have to do everything in it, and "up to" means at least 1.